### PR TITLE
Multimodal: Fix multimodal training example - tokenizer, Triton Cache Manager patch, docs

### PR DESCRIPTION
**Pull request by faradawn**
# What does this PR do ?

1. Currently, the tokenizer defaults to Huggingface tokenizer (Default Tokenizer) after loading the checkpoint. This PR enables multimodal tokenizer. Fix: https://github.com/NVIDIA/Megatron-LM/issues/3469
2. Currently, TRITON_CACHE_MANAGER patch is not needed in newer version and throws an error. This PR fixes: https://github.com/NVIDIA/Megatron-LM/issues/3239
3. Providing an example with 1 gpu (torchrun = 1) for better usability

## Contribution process

```mermaid
flowchart LR
    A[Pre-checks] --> B[PR Tests]
    subgraph Code Review/Approval
        C1[Expert Review] --> C2[Final Review]
    end
    B --> C1
    C2 --> D[Merge]
```

### Pre-checks

- [x] I want this PR in a versioned release and have added the appropriate Milestone (e.g., `Core 0.8`)
- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

The following process is enforced via the CODEOWNERS file for changes into `megatron/core`. For changes outside of `megatron/core`, it is up to the PR author whether or not to tag the Final Reviewer team.

<details>
<summary>For MRs into `main` branch</summary>

Feel free to message or comment the @mcore-oncall to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

#### (Step 2): Collect the expert reviewers reviews

1. Attach the `Expert Review` label when your PR is ready for review.
4. GitHub auto-assigns expert reviewers based on your changes. They will get notified and pick up your PR soon.

:warning: Only proceed to the next step once all reviewers have approved, merge-conflict are resolved and the CI is passing.  
Final Review might get declined if these requirements are not fulfilled.

#### (Step 3): Final Review

1. Add `Final Review` label
2. GitHub auto-assigns final reviewers based on your changes. They will get notified and pick up your PR soon.

#### (Optional Step 4): Cherry-pick into release branch

If this PR also needs to be merged into `core_r*` release branches, after this PR has been merged, select `Cherry-pick` to open a new PR into the release branch.

</details>

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### Merging your PR

Any member of [core-adlr](https://github.com/orgs/teams/NVIDIA/core-adlr) and [`core-nemo`](https://github.com/orgs/teams/NVIDIA/core-nemo) will be able to merge your PR.

### CODE REVIEW

File: examples/multimodal/pretrain_mistral_clip.sh [hunk 1]
```diff
@@ -92,7 +93,7 @@ OPTIONS=" \
     --eval-iters 10 \
     --eval-interval 1000 \
     --tokenizer-type MultimodalTokenizer \
-    --tokenizer-model mistralai/Mistral-7B-Instruct-v0.3 \
+    --tokenizer-model mistralai/Mistral-7B-v0.3 \
```
Comment (janEbert) [inline examples/multimodal/pretrain_mistral_clip.sh]: This is a bit icky; the `examples/multimodal/README.md` mentions downloading Mistral-7B-Instruct-v0.3, but the document it refers to (`docs/llama_mistral.md#mistral-7b`) uses Mistral-7B-v0.3.

I think we should keep this as Mistral-7B-Instruct-v0.3 and update `examples/multimodal/README.md` to mention more explicitly that `docs/llama_mistral.md#mistral-7b` links to a different model.

File: examples/multimodal/pretrain_mistral_clip.sh [hunk 2]
```diff
@@ -129,4 +130,4 @@ OPTIONS=" \
 export NVTE_APPLY_QK_LAYER_SCALING=0
 export NVTE_ALLOW_NONDETERMINISTIC_ALGO=${NONDETERMINISTIC_ATTN}
 
-torchrun --nproc_per_node 8 examples/multimodal/train.py ${OPTIONS}
\ No newline at end of file
+torchrun --nproc_per_node 1 examples/multimodal/train.py ${OPTIONS}
```
Comment (janEbert) [inline examples/multimodal/pretrain_mistral_clip.sh]: I don't have any stakes on whether this should be changed, but IMO, at the very least I'd keep this and `--tensor-model-parallel-size` at 4 to show the distribution capabilities. But perhaps someone else from @NVIDIA/multi-modal should weigh in here.

Also, could you please add a newline at the end of file to satisfy POSIX standards?

Comment (faradawn) [reply examples/multimodal/pretrain_mistral_clip.sh]: Restored tensor parallel size to 4. Added EOF new line!

File: examples/multimodal/pretrain_mistral_clip.sh (hunk 1 above)
Comment (faradawn) [reply examples/multimodal/pretrain_mistral_clip.sh]: Hi @janEbert , thanks for the suggestion - does the Instruct model work for multimodal? I have tested the based version, which is the in docs/llama_mistral.md#mistral-7b. How about I fix the README to all use the base version? Love to hear your thoughts

Comment (faradawn) [reply examples/multimodal/pretrain_mistral_clip.sh]: I take that back. I found the instruct model and added to the readme. Let me know if there is any issue.

Comment (janEbert) [reply examples/multimodal/pretrain_mistral_clip.sh]: Thank you so much!
